### DISCUSSION TITLE
2022.8 Bare metal examples have multiple directories and files

### DISCUSSION THREAD

**Discussion by Bharris100** [polarfire-soc/polarfire-soc-documentation#261] [Bare metal embedded software] [answered]
I am trying to update my drivers for my application starting with "mpfs-mmuart-interrrupt" BM application.  In my move to the 2022.8 example, I noticed that under boards/icicle-kit-es/platform_config there is a "linker" folder that contains just a readme.md and a "mpfs_hal_config" folder which contains a jenkinsfile, and an mss_sw_config.h file which does not match the same named file in "platform/platform_config_reference/mpfs_hal_config"  Are both mss_sw_config.h files used? And what is the correct configuration setup for both?

**Comment by hughbreslin** [comment, marked as answer]
Hey @Bharris100 you should use the sw_config.h file in your board folder - the idea of the folder structure is that there is a reference that you can copy and use to create a custom board, for example, "my-icicle-kit-es" with your custom configuration. There is a document which outlines the bare metal project structure [here](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/bare-metal-project-structure/bare-metal-software-project-structure.md) which may be useful :)

**Reply by misterhemi** [reply]
I just saw this post and tried the link to the document, it seems to be broken now.

**Reply by Bharris100** [reply]
@misterhemi check out this link
https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/bare-metal-embedded-software/bare-metal-software-project-structure.md

**Reply by misterhemi** [reply]
Thank you!
